MatrixSDKTests/MXClientInformationServiceUnitTests.swift

```diff
@@ -0,0 +1,165 @@
+//
+// Copyright 2022 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import XCTest
+@testable import MatrixSDK
+
+class MXClientInformationServiceUnitTests: XCTestCase {
+
+    func testUpdateData() {
+        MXSDKOptions.sharedInstance().enableNewClientInformationFeature = true
+
+        let mockDeviceId = "some_device_id"
+        let credentials = MXCredentials(homeServer: "", userId: "@userid:example.com", accessToken: "")
+        credentials.deviceId = mockDeviceId
+        guard let session = MXSession(matrixRestClient: MXRestClientStub(credentials: credentials)) else {
+            XCTFail("Failed to setup test conditions")
+            return
+        }
+
+        let service = MXClientInformationService(withSession: session)
+
+        let type = service.accountDataType(for: session)
+
+        // no client info before
+        let clientInfo = session.accountData.accountData(forEventType: type)
+        XCTAssertNil(clientInfo)
+
+        service.updateData()
+
+        // must be set after updateData
+        let updatedInfo1 = session.accountData.accountData(forEventType: type)
+        XCTAssertNotNil(updatedInfo1)
+
+        // try updating again
+        service.updateData()
+
+        // must not be changed
+        let updatedInfo2 = session.accountData.accountData(forEventType: type)
+        XCTAssertTrue(NSDictionary(dictionary: updatedInfo1!).isEqual(to: updatedInfo2))
```

Comment:
Is this test not passing if you remove this [guard](https://github.com/matrix-org/matrix-ios-sdk/pull/1588/files#diff-0f39a17e4f2198d23ebf5a26c9ba8f659ea7d87a8936b0329c5957895dd919b5R59) in the implementation? If I am not mistaken, even if the call is made, you would be comparing equal objects. Maybe it would be useful to create a session mock and inject it into `MXClientInformationService` and test that `setAccountData` is not called on it? In that case, you wold be able also to test the values passed. 
From test method name it is hard to understand what this test is about. I would suggest splitting this test into multiple tests. Something like:
- test_whenUpdateData_accountDataIsUpdatedWithCorrectValues
- test_whenUpdateDataSecondTime_setAccountDataIsNotCalled

Reply:
Updated.